### PR TITLE
Add support for DAP and lsp-treeview in scala layer.

### PR DESCRIPTION
**Pull request by kryptt**
Thank you for contributing to Spacemacs!

Before you submit this pull request, please ensure it is against the develop branch and not master.

Adding support for DAP and lsp-treeview in the scala layer

### CODE REVIEW

**Comment by kryptt** [comment]
@duianto This is working fine for me here. A note is that you should have metals up to date, to avoid complaining about treemacs

File: layers/+lang/scala/packages.el [hunk 1]
```diff
@@ -170,9 +166,6 @@
   (when spacemacs//scala-backend-ensime-p
     (add-hook 'scala-mode-hook #'spacemacs//scala-setup-ensime-flyspell)))
 
-(defun scala/init-noflet ()
```
[the remark this answers is not in the document]
Comment (kryptt) [reply layers/+lang/scala/packages.el:173]: it was used for decorating functions in a preview version of the metals plugin. at this point it is just left over cruft.

File: layers/+lang/scala/funcs.el [hunk 2]
```diff
@@ -52,7 +52,13 @@
 
 (defun spacemacs//scala-setup-metals ()
   "Setup LSP metals for Scala."
-  (add-hook 'scala-mode-hook #'lsp))
+  (add-hook 'scala-mode-hook #'lsp)
+  (add-hook 'lsp-mode-hook #'lsp-lens-mode))
```
[the remark this answers is not in the document]
Comment (kryptt) [reply layers/+lang/scala/funcs.el:56]: yeah, makes sense

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop.

**Pull request merged**